controller/src/main/java/io/pravega/controller/server/eventProcessor/ControllerEventProcessors.java

```diff
@@ -288,9 +300,67 @@ public boolean isReady() {
             streamMetadataTasks.initializeStreamWriters(clientFactory, config.getRequestStreamName());
             streamTransactionMetadataTasks.initializeStreamWriters(clientFactory, config);
             tableMetadataTasks.initializeStreamWriters(clientFactory, config.getKvtStreamName());
+
+            long delay = truncationInterval.get();
+            Futures.loop(this::isRunning, () -> Futures.delayedFuture(
```

Comment:
You need to add a check inside `truncate` to verify if `isRunning() == true`. Otherwise we can stop this service, but since this still has a 10-second delay, it will execute one call of `truncate` and then cancel it. This may not be the intended behavior.

Reply:
Added a Precondition on truncate method. 

Although its not really going to cause any incorrect behaviour even if an extra truncation ran (and would run until the executor is shutdown or curator is shutdown), as it is equivalent to us being in the middle of truncation when the service shutdown.